csrc/runtime/compiled_kernel.cpp

```diff
@@ -1078,9 +1079,14 @@ std::string _getStructuredCode(
 
   code += defineStdComplex();
   code += std::string("namespace ") + CompiledKernel::kernelNamespace() +
-      "{\n" + defineTypes() + defineIndexType(index_type) + kernelPreamble() +
-      "} // namespace " + CompiledKernel::kernelNamespace() + "\n";
+      "{\n" + defineTypes() + defineIndexType(index_type) + kernelPreamble();
 
+  if (has_cluster_reduction) {
+    code += nvfuser_resources::cluster_cu;
```

Comment:
Can we move this out of here and place it below like argsort_cu? I don't see any reason that we need to do this implicit insertion of namespace `nvf`. We haven't updated the existing ones yet, but the new ones like argsort.cu starts with `namespace nvf {`, and I think that's cleaner.